### ISSUE TITLE
Intent Assertion not working properly for irrelavant keywords

### ISSUE THREAD

**Issue by radhabasida** [codeforequity-at/botium-core#257]
Hi,

I am using Intent assertion in my script file using INTENT intentname.
If i am try keyword in watson it showing irrelavant but while using with botium, it is showing fail with reason that botium found another intent than expected. error should be like expected intent [intent name] but found nothing.

could you please help me out for this I am attaching the screenshot for the same.and also attaching convo files
[q_greeting.convo.txt](https://github.com/codeforequity-at/botium-core/files/3021612/q_greeting.convo.txt)
[q_greeting_input.utterances.txt](https://github.com/codeforequity-at/botium-core/files/3021613/q_greeting_input.utterances.txt)
![intent_assertion_issue](https://user-images.githubusercontent.com/48954471/55217995-d8bcd680-5226-11e9-9679-8135dd4fe435.png)
![intent_assertion_issue_1](https://user-images.githubusercontent.com/48954471/55217996-d8bcd680-5226-11e9-9e9b-7fa847e06c24.png)

**Comment by codeforequity-at** [comment]
Hello,
I understand what you mean, but that's not how the Watson connector is currently working. Botium uses the highest ranked intent for assertions, no matter how high or low the confidence is. The Watson user interface shows the dialog flow, which doesn't necessarily match the highest matched intent, for cases where the recognized confidence is too low, Watson chooses special "irrelevant" dialog pathes - recognized intent stays the same.

**Comment by radhabasida** [comment]
okay,
how botium can help me to identify those keywords for which irrelevant dialog comes. Also I have used INTENT_CONFIDENCE 100 te get the correct response and assertion result.

**Comment by codeforequity-at** [comment]
Intent resolution and dialog flow are two separate things in Watson. I think the only way to test the "Irrelevant" dialog pathes is by using utterances for assertions, not intent assertion.

**Comment by radhabasida** [comment]
so can botium help me for utterances assertion and if yes how can I use it?

**Comment by radhabasida** [comment]
Also if I can not verify whether correct intent identified or not, how can i make sure that my conversion is passed using botium. according to your resolution, I can never verify the utterances for which watson is giving irrelevant response because at the end dialogs are the ones using which we verify our keywords and train them according the response which we get after try it option

**Comment by codeforequity-at** [comment]
You can read about how Botium handles separation of dialog flow from the used utterances here: https://medium.com/@floriantreml/botium-in-a-nutshell-part-2-writing-chatbot-tests-8726204c8da6

We usually have two different training sets: one is for the intent resolution, we generate those test sets with the Botium CLI and update it from time to time - this is for asserting that continuous training doesn't break existing intent resolution. We work with our expected minimum confidence level here.

And another test set is for testing the dialog flow.
